src/form-builder/components/FormPrivilegeTable.jsx

```diff
@@ -0,0 +1,364 @@
+ this.handleAddRow = this.handleAddRow.bind(this);
+ this.handleTag = this.handleTag.bind(this);
+ this.formSave = this.formSave.bind(this);
+}
+componentWillMount(){
+this.fetchPrivileges();
+this.fetchFormData();
+}
+fetchFormData(){
+        const params =
+                    'v=custom:(id,uuid,name,version,published,auditInfo,' +
+                    'resources:(value,dataType,uuid))';
+            httpInterceptor
+                    .get(`${formBuilderConstants.formUrl}/${this.props.formUuid}?${params}`)
+                    .then((data) => {
+                      const parsedFormValue = data.resources.length > 0 ?
+                        JSON.parse(data.resources[0].value) : {};
+
+                      this.setState({   formData: data,
+                                        httpReceived: true,
+                                        loading: false,
+                                        formName: data.name,
+                                        referenceVersion: parsedFormValue.referenceVersion,
+                                        referenceFormUuid: parsedFormValue.referenceFormUuid
+                                    });
+
+
+                      //this.props.dispatch(formLoad(formControlsArray));
+                    })
+                    .catch((error) => {
+                      this.setErrorMessage(error);
+                      this.setState({ loading: false });
+                    });
+}
+  fetchPrivileges() {
+    let initialPrivileges = [];
+    const queryParams = `?=`;
+    const optionsUrl = `${formBuilderConstants.formPrivilegeUrl}${queryParams}`;
+     httpInterceptor.get(optionsUrl)
+        .then((initialPrivileges) => {
+               this.setState({ availablePrivileges: this.orderFormByVersion(initialPrivileges.results), loading: false });
+        })
+
+}
+ orderFormByVersion(privilege) {
+    const sampleList=[];
+     privilege.forEach((privilege) => {
+       const item ={
+                 value:privilege.display,
+                 uuid: privilege.uuid,
+                 label:privilege.display,
+                 }
+       sampleList.push(item);
+       });
+    return sampleList;
+  }
+
+
+ handleTag(event,idx) {
+  const formPrivileges = this.state.formPrivileges.slice();
+  const availablePrivileges = this.state.availablePrivileges.slice();
+        const tempItem = {
+             uuid: event.uuid,
+            privilegeName: event.display,
+            isEditable: false,
+            isViewable: false,
+          };
+
+       if(event.event != undefined && (formPrivileges.length >0) && (event.event.currentTarget.type === "checkbox")){
+
+            if(event.event.currentTarget.name === 'isEditable'){
+                formPrivileges[event.idx].isEditable = event.event.currentTarget.checked;
+            }
+            if(event.event.currentTarget.name === 'isViewable'){
+                formPrivileges[event.idx].isViewable = event.event.currentTarget.checked;
+            }
```

Comment:
@buvaneswari-arun what are the other options **event.event.currentTarget.name** has apart from what mentioned above ? can we combine these 2 if conditions ?